File: tests/fixtures/sloppy.ts
import axios from 'axios';

// fetch the data
async function fetchData(url: string): Promise<any> {
    const response = await axios.get(url);
    const data: any = response.data;
    return data;
}

// process the data
function processData(items: any[]): any[] {
    return items.map((item: any) => {
        const result: any = {
            name: item.name,
            value: item.value,
        };
        return result;
    });
}

// handle the request
async function handleRequest(req: any, res: any): Promise<void> {
    try {
        const data = await fetchData("https://api.example.com/data");
        const processed = processData(data as any);
        console.log("processed data:", processed);
        console.log("items count:", processed.length);
        console.log("first item:", processed[0]);
        res.json(processed);
    } catch (error) {
        console.error("Error:", error);
        console.log("Failed to process request");
        res.status(500).json({ error: "Internal server error" });
    }
}

// get the result
function getData(): Promise<object> {
    return fetch("/api/data")
        .then(res => res.json())
        .then(data => data.items)
        .then(items => items.filter((i: any) => i.active))
        .catch(() => {});
}

// @ts-ignore
const legacyValue = window.oldAPI.getValue();
// @ts-ignore
const anotherLegacy = window.oldAPI.process();
// @ts-expect-error
const brokenThing: number = "not a number";

const status = true;
const label = status ? "active" : status === false ? "inactive" : status === null ? "unknown" : "error";

// TODO: Add error handling
// TODO: Implement this
// TODO: Add validation

File: tests/fixtures/sloppy.py
import os
import sys
import json
import logging
from typing import *
from collections import *

# initialize the logger
logger = logging.getLogger(__name__)

# process the data
def process_data(data):
    result = []
    for item in data:
        result.append(item)
    return result

# handle the request
def handle_request(request):
    try:
        data = json.loads(request.body)
        result = process_data(data)
        print(f"Processed {len(result)} items")
        print(f"Result: {result}")
        print(f"Done processing")
        return result
    except:
        print("Error occurred")
        return None

# get the result
def get_result(path: str) -> dict:
    try:
        with open(path) as f:
            data = json.load(f)
        return data
    except Exception as e:
        print(f"Failed: {e}")
        return {}

# save the data
def save_data(data, path):
    with open(path, 'w') as f:
        json.dump(data, f)

def load_data(path) -> list:
    with open(path) as f:
        return json.load(f)

def fetch_data(url: str):
    import requests
    response = requests.get(url)
    return response.json()

def execute(cmd):
    os.system(cmd)
    print(f"Executed: {cmd}")
    print(f"Done")
